### PR TITLE
Fix custom data_dir

### PR DESCRIPTION
**Pull request by phantomdata**
This PR fixes the previous implementation for custom `data_dir` support.

- Converts hard-coded "data" paths to dynamic references to `ConrefFS.data_dir_name`
- Adds `ConrefFS.data_dir_name` so that it can be called at any point in the Nanoc pipeline (such as in the Rules file).
- Fixes incorrect key lookup in previous implementation

### CODE REVIEW

File: lib/nanoc-conref-fs/datafiles.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def self.apply_conditionals(config, path:, content:, rep:)
       end
 
       path = path.dup
-      path.slice!('data/')
+      path.slice!("#{ConrefFS.data_dir_name(config)}/") # Beware the slashes, they are important for tokenization
```
Comment (gwincr11) [inline lib/nanoc-conref-fs/datafiles.rb]: Should the slash come from the data_dir_name function, it seems like it is also added in other places this is used...

File: lib/nanoc-conref-fs/conref-fs.rb [hunk 2]
```diff
@@ -102,12 +98,32 @@ def parse_with_frontmatter(content_filename)
...
+  # - An optional already-loaded config, so that this can be used later in the
+  #   Nanoc pipeline.
+  #
+  # Returns:
+  # - Custom `data_dir` attribute from the 'conref-fs' data_source
+  # - Default `data_dir` of 'data' if none is configured in `nanoc.yaml`
+  def self.data_dir_name(config=nil)
+    config ||= YAML.load_file('nanoc.yaml')
+    data_sources = config.fetch("data_sources") { [] }
+    data_source = data_sources.find { |ds| ds["type"] == "conref-fs" } || { "data_dir": "data" }
```
Comment (gwincr11) [inline lib/nanoc-conref-fs/conref-fs.rb]: I am not really a huge fan of this syntax, I find it hard to understand.

Comment (phantomdata) [reply lib/nanoc-conref-fs/conref-fs.rb]: :100:  Converted to the way I like it too.  :)

File: lib/nanoc-conref-fs/datafiles.rb (hunk 1 above)
Comment (phantomdata) [reply lib/nanoc-conref-fs/datafiles.rb]: Yuuup.  So, this actually opens up a whole can of worms that I don't think are important at this time.  I spent about 45 minutes just now rolling through and there are some tests that are doing some unique things with manipulating the data_dir, and I can't predict what else in the system might be depending on this.

Since my primary goal here was to get the custom `data_dir` working so that I could move forward with another project that's dependent on it, I'd like to table this for the time being.  :)

**Comment by phantomdata** [comment]
So I've been testing this with our project locally.  Since this is subject to gemception (project1 depends on project2 depends on nanoc-conref-fs)... I'd love to pair up with you and see how you'd like to do it more publicly.  Shall we coordinate over teh Slacks?

File: lib/nanoc-conref-fs/conref-fs.rb [hunk 3]
```diff
@@ -10,22 +13,18 @@ class ConrefFS < Nanoc::DataSources::Filesystem
...
-  def data_dir_name
-    config.fetch(:data_dir) { |_| 'data' }
-  end
-
   def self.load_data_folder(config, rep)
     return unless NanocConrefFS::Variables.variables[rep].nil?
     data_files = NanocConrefFS::Variables.data_files
     data = NanocConrefFS::Datafiles.process(data_files, config, rep)
-    NanocConrefFS::Variables.variables[rep] = { 'site' => { 'config' => config, 'data' => data } }
+    NanocConrefFS::Variables.variables[rep] = { 'site' => { 'config' => config, data_dir_name => data } }
```
Comment (gwincr11) [inline lib/nanoc-conref-fs/conref-fs.rb]: Why did you need to change this to data_dir_name?

Comment (phantomdata) [reply lib/nanoc-conref-fs/conref-fs.rb]: Half of the chain is based on the file path, the other is based on the hash path.  The two intersect and over-lap throughout the library (using the file path to lookup values in the hash) and must be made to be identical.  :/

**Comment by gwincr11** [comment]
@gjtorikian would you mind giving this a 👀 ?
